features/lockfiles/ci/build.py

```diff
@@ -0,0 +1,61 @@
+import os, json, shutil
+
+def run(cmd):
+    ret = os.system(cmd)
+    if ret != 0:
+        raise Exception("Failed command: %s" % cmd)
+
+def load(filename):
+    with open(filename, "r") as f:
+        return f.read()
+
+def rm(path):
+    if os.path.isfile(path):
+        os.remove(path)
+    elif os.path.isdir(path):
+        shutil.rmtree(path)
+
+rm("release")
+rm("build_server_folder")
+rm("bo.json")
+run('conan remove "*" -f')
+
+run("conan config set general.default_package_id_mode=full_version_mode")
+run("cd PkgZ && conan create . PkgZ/0.1@user/testing")
+run("cd PkgA && conan create . PkgA/0.1@user/testing")
+run("cd PkgB && conan create . user/testing")
+run("cd PkgC && conan create . user/testing")
+run("cd App && conan create . user/testing")
+
+# Freeze the App dependencies in a lockfile in the release subfolder
+print("\n********** Freeze dependencies in a lockfile ***********")
+run("conan graph lock App/0.1@user/testing --lockfile=release")
+
+# A new version of Z will not affect at all
+run("cd PkgZ && conan create . PkgZ/0.2@user/testing")
+# There is a change in A, a new version that we want to test
+print("\n********** Creating PkgA/0.2 ***********")
+run("cd PkgA && conan create . PkgA/0.2@user/testing --lockfile=../release")
+
+print("\n********** Computing the build order after PkgA/0.2 ***********")
+run("conan graph build-order ./release --json=bo.json --build=missing")
+
+build_order = json.loads(load("bo.json"))
+
+while build_order:
```

Comment:
Maybe a comment about the first level in the build order could be launched in parallel.